### PR TITLE
Add business rule to sign contract

### PR DESCRIPTION
**Pull request by meaboutsoftware**
This PR includes work related to:

- introduction of a business rule that checks if contract can be signed or not
- integration of the business rule in contract entity on signing action
- fix to cancellation token passing in 2 endpoints
- moving usings to a namespace

### CODE REVIEW

File: Src/Fitnet.IntegrationTests/Contracts/SignContract/SignContractRequestFaker.cs [hunk 1]
```diff
@@ -1,13 +1,13 @@
-using SuperSimpleArchitecture.Fitnet.Contracts.SignContract;
-
 namespace SuperSimpleArchitecture.Fitnet.IntegrationTests.Contracts.SignContract;
 
+using SuperSimpleArchitecture.Fitnet.Contracts.SignContract;
+
 internal sealed class SignContractRequestFaker : Faker<SignContractRequest>
 {
-    public SignContractRequestFaker()
+    public SignContractRequestFaker(DateTimeOffset signedAt)
```
Comment (kamilbaczek) [inline Src/Fitnet.IntegrationTests/Contracts/SignContract/SignContractRequestFaker.cs:7]:
```suggestion
    public SignContractRequestFaker(DateTimeOffset signedAt) =>
        CustomInstantiator(_ =>
            new SignContractRequest(signedAt)
        );```

File: Src/Fitnet.IntegrationTests/Contracts/SignContract/SignContractTests.cs [hunk 2]
```diff
@@ -61,4 +61,9 @@ private async Task<Guid> PrepareContract()
     }
 
     private static string BuildUrl(Guid id) => ContractsApiPaths.Sign.Replace("{id}", id.ToString());
+    
+    private static DateTimeOffset GetValidSignedAtDate()
```
Comment (kamilbaczek) [inline Src/Fitnet.IntegrationTests/Contracts/SignContract/SignContractTests.cs:65]:
```suggestion
    private static DateTimeOffset GetValidSignedAtDate() => 
        DateTimeOffset.Now.AddDays(1).ToUniversalTime();
```

File: Src/Fitnet.UnitTests/Contracts/SignContract/BusinessRules/ContractCanOnlyBeSignedWithin30DaysFromPreparationTests.cs [hunk 3]
```diff
@@ -0,0 +1,53 @@
+using SuperSimpleArchitecture.Fitnet.Contracts.SignContract.BusinessRules;
+using SuperSimpleArchitecture.Fitnet.Shared.BusinessRulesEngine;
+
+namespace SuperSimpleArchitecture.Fitnet.UnitTests.Contracts.SignContract.BusinessRules;
+
+public sealed class ContractCanOnlyBeSignedWithin30DaysFromPreparationTests
+{
+    [Fact]
+    public void Given_signed_at_date_which_is_more_than_30_days_from_prepared_at_date_Then_validation_should_throw()
```
Comment (kamilbaczek) [inline Src/Fitnet.UnitTests/Contracts/SignContract/BusinessRules/ContractCanOnlyBeSignedWithin30DaysFromPreparationTests.cs:9]: Could we make test parameterized. 
to be able to change test data like 31 without changing of the test body?

File: Src/Fitnet.UnitTests/Contracts/SignContract/BusinessRules/ContractCanOnlyBeSignedWithin30DaysFromPreparationTests.cs [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+    [Fact]
+    public void Given_signed_at_date_which_is_less_than_30_days_from_prepared_at_date_Then_validation_should_pass()
+    {
+        // Arrange
+
+        // Act
+        var act = () =>
+            BusinessRuleValidator.Validate(
+                new ContractCanOnlyBeSignedWithin30DaysFromPreparation(DateTimeOffset.Now,
+                    DateTimeOffset.Now.AddDays(29)));;
```
Comment (kamilbaczek) [inline Src/Fitnet.UnitTests/Contracts/SignContract/BusinessRules/ContractCanOnlyBeSignedWithin30DaysFromPreparationTests.cs:48]: Please check this class there is two places with ;; instead ;

**Review by kamilbaczek — CHANGES_REQUESTED** [review]
Could you change a few minor suggestions?

File: Src/Fitnet.IntegrationTests/Contracts/SignContract/SignContractRequestFaker.cs (hunk 1 above)
[the remark this answers is not in the document]
Comment (meaboutsoftware) [reply Src/Fitnet.IntegrationTests/Contracts/SignContract/SignContractRequestFaker.cs:7]: Done

File: Src/Fitnet.IntegrationTests/Contracts/SignContract/SignContractTests.cs (hunk 2 above)
Comment (meaboutsoftware) [reply Src/Fitnet.IntegrationTests/Contracts/SignContract/SignContractTests.cs:65]: Done

File: Src/Fitnet.UnitTests/Contracts/SignContract/BusinessRules/ContractCanOnlyBeSignedWithin30DaysFromPreparationTests.cs (hunk 4 above)
Comment (meaboutsoftware) [reply Src/Fitnet.UnitTests/Contracts/SignContract/BusinessRules/ContractCanOnlyBeSignedWithin30DaysFromPreparationTests.cs:48]: Done, good spot!

File: Src/Fitnet.UnitTests/Contracts/SignContract/BusinessRules/ContractCanOnlyBeSignedWithin30DaysFromPreparationTests.cs (hunk 3 above)
Comment (meaboutsoftware) [reply Src/Fitnet.UnitTests/Contracts/SignContract/BusinessRules/ContractCanOnlyBeSignedWithin30DaysFromPreparationTests.cs:9]: In this particular case I am more into direction of having separate test cases without parametrisation. The reason is that:

- there are 3 test cases
- each test checks concrete border case
- 2 tests passes, 1 test throws (different expectations)

Usually I am for parametrisation, however here I would vote to do it this way (it is more readable)

**Review by kamilbaczek — APPROVED** [review]
Quick fixes! Thanks

**Review by kamilbaczek — APPROVED** [review]
Good job!

**Pull request merged**